packages/grafana-ui/src/components/Table/FilterList.tsx

```diff
@@ -2,33 +2,131 @@ import { css, cx } from '@emotion/css';
 import React, { useCallback, useMemo, useState } from 'react';
 import { FixedSizeList as List } from 'react-window';
 
-import { GrafanaTheme2, SelectableValue } from '@grafana/data';
+import { GrafanaTheme2, formattedValueToString, getValueFormat, SelectableValue } from '@grafana/data';
 
-import { Checkbox, FilterInput, Label, VerticalGroup } from '..';
+import { ButtonSelect, Checkbox, FilterInput, HorizontalGroup, Label, VerticalGroup } from '..';
 import { useStyles2, useTheme2 } from '../../themes';
 
 interface Props {
   values: SelectableValue[];
   options: SelectableValue[];
   onChange: (options: SelectableValue[]) => void;
   caseSensitive?: boolean;
+  showOperators?: boolean;
 }
 
 const ITEM_HEIGHT = 28;
 const MIN_HEIGHT = ITEM_HEIGHT * 5;
 
-export const FilterList = ({ options, values, caseSensitive, onChange }: Props) => {
+const operatorSelectableValue = (op: string): SelectableValue<string> => {
+  const result: SelectableValue<string> = { label: op, value: op };
+  switch (op) {
+    case 'Contains':
+      result.description = 'Contains';
+      break;
+    case '=':
+      result.description = 'Equals';
+      break;
+    case '!=':
+      result.description = 'Not equals';
+      break;
+    case '>':
+      result.description = 'Greater';
+      break;
+    case '>=':
+      result.description = 'Greater or Equal';
+      break;
+    case '<':
+      result.description = 'Less';
+      break;
+    case '<=':
+      result.description = 'Less or Equal';
+      break;
+    case 'Expression':
+      result.description =
+        'Bool Expression (Char v represents the column value in the expression, e.g. "v >= 10 && v <= 12")';
+      break;
+  }
+  return result;
+};
+const OPERATORS = ['Contains', '=', '!=', '<', '<=', '>', '>=', 'Expression'].map(operatorSelectableValue);
+const REGEX_OPERATOR = OPERATORS.filter((op) => op.value === 'Contains')[0];
+const XPR_OPERATOR = OPERATORS.filter((op) => op.value === 'Expression')[0];
+
+const comparableValue = (value: string): string | number | Date | boolean => {
+  value = value.trim().replace(/\\/g, '');
+
+  // Does it look like a Date (Starting with pattern YYYY-MM-DD* or YYYY/MM/DD*)?
+  if (/^(\d{4}-\d{2}-\d{2}|\d{4}\/\d{2}\/\d{2})/.test(value)) {
```

Comment:
This regex doesn't have an end of string match `$`

Reply:
the comment does say it's aiming for "startsWith" behavior tho...

Reply:
Yes, the idea is to match text if they are representing dates (e.g. YYYY-MM-DD), and datetimes as well (of example YYYY-MM-DD HH::MM:SS).

Reply:
Hi @codeincarnate 
> Other than that, I think that the control should be moved below "Select All Values" with a heading label of "Filter Available Values" or similar. 
 
This control only filters in the panel a list of values that you can finally select by hand or with other control. Other recent PR was merged in 10.3.x adding new control in the bottom to select or unselect all of that displayed values.

![image](https://github.com/grafana/grafana/assets/5576272/a02d06c1-9e77-4723-9024-31f04a2c62b4)

Maybe it seems is strange you are changing this list of displayed values located in the top with a control located in the bottom.

Reply:
Of course, if you are thinking it is better to move that control to the position you say, I am happy to change that.

Reply:
> Of course, if you are thinking it is better to move that control to the position you say, I am happy to change that.

No worries, I think it's fine for now 😄